### PR TITLE
Added the research custom post type

### PR DESCRIPTION
**Pull request by jmbarne3**
**Description**
Adds the `research` custom post type to the plugin. The default menu names may get changed moved going forward, but I used what makes the most sense to me to begin with.

**Motivation and Context**
The primary purpose of the plugin is to provide of the plugin is to provide this post type.

**How Has This Been Tested?**
We don't have a dev environment for this site yet. Functionality can be verified by pulling down the plugin and activating/deactivating it.

**Types of changes**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

**Checklist:**

- [x] My code follows the code style of this project.
- [x] My change requires an update to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by rjucf — APPROVED** [review]

**Review by cadie — APPROVED** [review]

File: includes/class-post-type.php [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+class Research {
+	private
+		/**
+		 * @var string The singular label for the post type
+		 */
+		$singular,
+		/**
+		 * @var string The lowercase variant of $singular
+		 */
+		$singuarl_lower,
```
Comment (cjg89) [inline includes/class-post-type.php]: Typo "singuarl"

File: includes/class-post-type.php [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+	/**
+	 * Appends the post meta to the `meta`
+	 * field on each post.
+	 * @author Jim Barnes
+	 * @since 1.0.0
+	 * @param WP_Post[] $posts The post array
+	 * @param WP_Query $query The query object. Added for convenience
+	 * @return WP_Post[] The post array
+	 */
+	public function meta( $posts, $query ) {
```
Comment (cjg89) [inline includes/class-post-type.php]: Do we still need to be appending meta data to post objects manually?  IIRC post meta should be accessible as post object properties already: https://developer.wordpress.org/reference/functions/get_post_meta/#comment-1894

**Review by cjg89 — APPROVED** [review]

File: includes/class-post-type.php (hunk 1 above)
Comment (jmbarne3) [reply includes/class-post-type.php]: Corrected

File: includes/class-post-type.php (hunk 2 above)
Comment (jmbarne3) [reply includes/class-post-type.php]: Removed.

**Pull request merged**